noxfile.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import nox
+
+
+ALL_PYTHONS = ['3.7', '3.8', '3.9']
+
+nox.options.sessions = ['lint', 'type', 'tests']
+
+
+@nox.session(reuse_venv=True)
+def lint(session):
+    """
+    Run the linter.
+    """
+    session.install('pre-commit')
+    session.run('pre-commit', 'run', '--all-files', *session.posargs)
+
+
+@nox.session(python=ALL_PYTHONS, reuse_venv=True)
+def tests(session):
+    """
+    Run the unit and regular tests.
+    """
+    session.install(
+        'tomli',
+        'packaging',
+        'rich',
+        'wheel',
+        'pytest',
+        'pytest-cov',
+        'pytest-mock',
+        'backports.cached-property',
+        'pep621',
+    )
+    session.run('pytest', '--showlocals', '-vv', '--durations=1', *session.posargs)
```

Comment:
nitpick: could you use this instead?

https://github.com/FFY00/dbus-objects/blob/6aceac3c832087515ebf5aae7ff6b9678aea9565/noxfile.py#L22-L32